### PR TITLE
Update to modern stack

### PR DESCRIPTION
**Pull request by thibaudcolas**
This is quite a bit of work, although most of it is copy-paste from [Draftail](https://github.com/springload/draftail). I initially thought about switching from the rackt cli to [nwb](https://github.com/insin/nwb/), but I'd rather have us know how to use Webpack / ESLint / Jest directly than use tools that hide them. If there was an official "create-react-component" CLI that could have been another story.

Tooling & metadata:

- Switch build to webpack
- Add Editorconfig & nvmrc
- Add git hooks
- Clean up package.json metadata
- Update license and changelog file to use more common format
- Move ESLint config to package.json, using eslint-config-springload
- Update Travis config to notify on build & collect coverage
- Add GA tracking to example page
- Update basic tests to use Jest
- Remove UMD build from GitHub
- Add basic deploy to gh-pages script

Component:

- Remove unnecessary animationend vendor prefixing
- Convert all code to be compatible with React 15.5
- Update peerDependencies to reflect dependency on prop-types

### CODE REVIEW

**Comment by thibaudcolas** [comment]
Merging this as-is so we can benefit from the better tooling. @samanthaksanders @vincentaudebert please review anyway!

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+6
```
Comment (vincentaudebert) [inline .nvmrc:1]: Already outdated? 😂

File: lib/index.test.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    });
+
+    it('#isOpen', () => {
+        expect(shallow((
+            <Modal isOpen={true}>
+                Test
+            </Modal>
+        ))).toMatchSnapshot();
+    });
+});
```
Comment (vincentaudebert) [inline lib/index.test.js:21]: Very poor coverage (<25%). I know it's not the purpose of this PR though
I will see if I can improve this :)

File: package.json [hunk 3]
```diff
@@ -1,51 +1,94 @@
...
   "repository": {
     "type": "git",
     "url": "https://github.com/springload/react-accessible-modal.git"
   },
-  "homepage": "https://github.com/springload/react-accessible-modal",
-  "bugs": "https://github.com/springload/react-accessible-modal/issues",
-  "directories": {
-    "example": "examples"
+  "eslintConfig": {
+    "extends": "springload"
```
Comment (vincentaudebert) [inline package.json:20]: WOW! Love it!

File: .nvmrc (hunk 1 above)
Comment (thibaudcolas) [reply .nvmrc:1]: We only use LTS versions of node (4, 6, 8, 10 in the future), so Node 6 will still be supported until December 2019.

**Pull request merged**